### PR TITLE
Add new Jenkins CR job interface and fixes/improvements to current dashboard

### PR DESCRIPTION
**Pull request by ali222**
These changes will let for stand-alone Jenkins CR job execution as well as updating clients' CR jobs remotely.
Also, there are several fixes and improvements to current Jenkins dashboard.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java [hunk 1]
```diff
@@ -596,9 +712,1623 @@ public String getDeployHistory() {
...
+       Connection conn = null;
+       Statement stat = null;
+	   
+	   String someString = new String();
+	  	   
+	   System.out.println(getCurentDateTime() + ": Getting the user executing Jenkins...");
+	   String user = System.getProperty("user.name");
+	   System.out.println(user);
+	   
+	   conn = CustomDBConnection.getConnection("mycomputer", "1433", "test", getdbUser(), getdbPassword(), getSQLauth());
```
Comment (tmcmains) [inline src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java]: what is this for?

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java [hunk 2]
```diff
@@ -596,9 +712,1623 @@ public String getDeployHistory() {
...
+	   
+	   System.out.println(SQL);
+	   
+	   
+	   
+	   //conn = CustomDBConnection.getConnection("mycomputer", "1433", "test", getdbUser(), getdbPassword(), getSQLauth());
+	   //String SQL = "select * from dbo.persons where name = 'john';";
+	   //String SQL = "select * from dbo.persons;";
+	   
+	   conn = CustomDBConnection.getConnection(server, "1433", "placeholderForDB", getdbUser(), getdbPassword(), getSQLauth());
```
Comment (tmcmains) [inline src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java]: why placeholderForDB?

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java [hunk 3]
```diff
@@ -596,9 +712,1623 @@ public String getDeployHistory() {
...
+	   
+	   
+	   
+	   
+	   
+	   //conn = CustomDBConnection.getConnection("mycomputer", "1433", "test", getdbUser(), getdbPassword(), getSQLauth());
+	   //String SQL = "select * from dbo.persons where name = 'john';";
+	   //String SQL = "select * from dbo.persons;";
+	   
+	   conn = CustomDBConnection.getConnection(server, "1433", "placeholderForDB", getdbUser(), getdbPassword(), getSQLauth());
```
Comment (tmcmains) [inline src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java]: placeholderForDB?

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java [hunk 4]
```diff
@@ -596,9 +712,1623 @@ public String getDeployHistory() {
...
+			   step_name = step_name.replace("'","''");
+			   command = command.replace("'","''");
+			   
+			   
+			   String SQL = "USE msdb;\n" +
+					"EXEC dbo.sp_add_jobstep  \n" +
+					"    @job_name = N'" + job_name + "',  \n" +
+					"    @step_name = N'" + step_name + "', \n" +
+					"	@subsystem = N'" + subsystem + "', \n" +
+					"	@command = N'" + command + "', \n" +
```
Comment (tmcmains) [inline src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java]: will we have any issues with special characters in string?

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java [hunk 5]
```diff
@@ -596,9 +712,1623 @@ public String getDeployHistory() {
...
+	   
+	   //conn = CustomDBConnection.getConnection("mycomputer", "1433", "msdb", getdbUser(), getdbPassword(), getSQLauth());
+	   //String SQL = "EXEC dbo.sp_help_job @job_name = N'" + job + "',  @job_aspect = N'steps';";
+	   //System.out.println(getCurentDateTime() + ": Here is the built change request:");
+	   //System.out.println(SQL);
+	   
+	   //conn = CustomDBConnection.getConnection("TESTSQLTST04", "1433", "test_warehouse_dev04", getdbUser(), getdbPassword(), getSQLauth());
+	   //String SQL = "select age_range_id, age_range from dbo.age_range where age_range_id = 1;";
+	   
+	   conn = CustomDBConnection.getConnection(activeServer, "1433", "placeholderForDB", getdbUser(), getdbPassword(), getSQLauth());
```
Comment (tmcmains) [inline src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java]: placeholderForDB?

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java (hunk 1 above)
Comment (ali222) [reply src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java]: I used getTestDataFromLocalSQLserver function just for testing/debugging purposes initially. getTestDataFromLocalSQLserver function is not used/called anywhere in the code. Do you want me to remove it?

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java (hunk 2 above)
Comment (ali222) [reply src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java]: I thought I would need to specify database name explicitly in some cases when I started development and just kept the placeholderForDB as a dummy value which is mapped to db parameter in getConnection function. Eventually, I didn't really need it but there could be cases in the future where I may need to leverage it. Do you want me to remove unused db parameter from getConnection function for now?

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java (hunk 3 above)
Comment (ali222) [reply src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java]: Please see explanation above for placeholderForDB value

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java (hunk 4 above)
Comment (ali222) [reply src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java]: I didn't take this use case into consideration. Let me follow up on that.

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java (hunk 5 above)
Comment (ali222) [reply src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java]: Please see explanation above for placeholderForDB value

File: src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java (hunk 4 above)
Comment (ali222) [reply src/main/java/org/jenkinsci/plugins/environmentdashboard/EnvDashboardView.java]: We should be fine. Just tested below command. It's parsed successfully as well as saved via Jenkins CR job interface:
select * from persons where name = '/][][%$^?\.!;"'

**Pull request merged**